app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -114,6 +116,13 @@ import java.lang.ref.WeakReference
 @SuppressWarnings("TooManyFunctions", "LargeClass")
 open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
+    init {
+        // we need to store startup timestamp for warm startup. we cant directly store
+        // inside AppStartupTelemetry since that class lives inside components and
+        // components requires context to access. we store it as a static member
+        AppLaunchTimeMeasurement.homeActivityInitTimeStampNano = SystemClock.elapsedRealtimeNanos()
```

Comment:
So I'm not a fan  of static variables, not sure how feasible this is for you, but how about getting the `init` time in here and passing it when we have access to `components` ?